File: uva/uva10013.cpp
#include<iostream>
using namespace std;
int num1[1000005];
int num2[1000005];
int carry[1000005];
int ans[1000005];
int main(){
    int nn,m;
    cin>>nn;
    bool f=true;
    while(nn--){
        cin>>m;
        for(int i=m-1;i>=0;i--)
            cin>>num1[i]>>num2[i];
        for(int i=0;i<m;i++){
            //cout<<(carry[i]+num1[i]+num2[i])<<endl;
            ans[i]=(carry[i]+num1[i]+num2[i])%10;
            carry[i+1] = (carry[i]+num1[i]+num2[i])/10;
        }
        if(f){
            f=false;
        }else{
            cout<<endl;
        }
        for(int i=m-1;i>=0;i--){
            cout<<ans[i];
        }
        cout<<endl;

    }
}

File: uva/uva10013.java
import java.util.*;
import java.io.*;
import java.math.BigInteger;
public class uva10013{
    public static void main(String [] args){
        Scanner in = new Scanner(System.in);
        int N;
        N=in.nextInt();
        for(int i=0;i<N;i++){
            int M = in.nextInt();
            String num1 = new String(""), num2 = new String("");
            int dig1,dig2;
            for(int k=0;k<M;k++){
                dig1 = in.nextInt();
                dig2 = in.nextInt();
                num1+=dig1;
                num2+=dig2;
            }
            BigInteger n1 = new BigInteger(num1);
            BigInteger n2 = new BigInteger(num2);
            System.out.println(n1.add(n2));
            if (i != N-1) {
                System.out.println("");
            }
        }
    }
}